utils/utils.go

```diff
@@ -47,6 +47,11 @@ func CustomizeDeployment(deploy *appsv1.Deployment, ba common.BaseComponent) {
 		}
 	}
 
+	dp := ba.GetDeployment()
+	if dp.GetDeploymentUpdateStrategy() != nil {
```

Comment:
wondering if a nil check on `dp` is needed since `spec.deployment` might not be set in the CR.